src/requests.c

```diff
@@ -117,22 +117,16 @@ static size_t resp_callback(char *content, size_t size, size_t nmemb,
                             req_t *userdata)
 {
     size_t real_size = size * nmemb;
-
-    /* extra 1 is for NULL terminator */
-    userdata->text = realloc(userdata->text, userdata->size + real_size + 1);
+	long original_userdata_size = userdata->size;
+	
+	/* extra 1 is for NULL terminator */
+    userdata->text = (char*) realloc(userdata->text, userdata->size + real_size + 1);
     if (userdata->text == NULL)
         return -1;
 
     userdata->size += real_size;
-
-    /* create NULL terminated version of `content' */
-    char *responsetext = strndup(content, real_size + 1);
-    if (responsetext == NULL)
-        return -1;
-
-    strncat(userdata->text, responsetext, real_size);
-
-    free(responsetext);
+	/* concatenate userdata->text with the response content */
+	memcpy(userdata->text + original_userdata_size, content, real_size);
```

Comment:
Since we're not using strncat anymore (presumably to support binary data), we need to remember to NULL-terminate the string manually. I believe it would look like this:

```suggestion
	memcpy(userdata->text + original_userdata_size, content, real_size);
	userdata->text[original_userdata_size + real_size] = '\0';
```

Reply:
Indeed, that seems a bit better :)